File: DataAccess/Entities/Product.cs
using Newtonsoft.Json;

namespace DataAccess.Entities
{
    public class Product: BaseEntity
    {
        public string IdExternal { get; set; }

        public string? Name { get; set; }

        public string? ProductQuantity { get; set; }        
        
        public string? ProductQuantityUnit { get; set; }

        public string? ServingQuantity { get; set; }

        public string? ServingQuantityUnit { get; set; }
        public Nutriments? Nutriments { get; set; }
    }
}

File: fodmaplog.client/src/app/domain/Product.ts
import { Nutriments } from "./Nutriments";

export class Product {
    id: number;
    name: string;
    productQuantity: string;
    productQuantityUnit: string;
    servingQuantity: number;
    servingQuantityUnit: string;
    nutriments: Nutriments;

    constructor(id: number, name: string, nutriments: Nutriments,
        productQuantity: string, productQuantityUnit: string, servingQuantity: number, servingQuantityUnit: string) {
        this.id = id;
        this.name = name;
        this.nutriments = nutriments;
        this.productQuantity = productQuantity;
        this.productQuantityUnit = productQuantityUnit;
        this.servingQuantity = servingQuantity;
        this.servingQuantityUnit = servingQuantityUnit;
    }
}
